skywalking/plugins/sw_aiohttp.py

```diff
@@ -0,0 +1,65 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from skywalking import Layer, Component
+from skywalking.trace import tags
+from skywalking.trace.context import get_context
+from skywalking.trace.tags import Tag
+
+
+def install():
+    from aiohttp import ClientSession
+    from multidict import CIMultiDict, MultiDict, MultiDictProxy
+    from yarl import URL
+
+    async def _sw_request(self: ClientSession, method: str, str_or_url, **kwargs):
+        url = URL(str_or_url)
+        peer = \
+            (url.scheme + '://' if url.scheme else '') + \
+            ((url.user or '') + ':' + (url.password or '') + '@' if url.user or url.password else '') + \
```

Comment:
I don’t think we want to include `password` here

Reply:
Yes its probably better not to. The thing is I copied this behavior from the other plugins so they will need to be changed to explicitly remove the password as well since `urllib.parse` always returns it as part of `netloc` if it is present and so would be recorded. Then the Node agent also, as well I am wondering if the Java plugins record this?

Reply:
> Yes its probably better not to. The thing is I copied this behavior from the other plugins so they will need to be changed to explicitly remove the password as well since `urllib.parse` always returns it as part of `netloc` if it is present and so would be recorded. Then the Node agent also, as well I am wondering if the Java plugins record this?

The Java agent doesn't record this

Reply:
> The Java agent doesn't record this

So just Python and Node, both incoming and outgoing, http and any other protocols (layers)?

Reply:
> > The Java agent doesn't record this
> 
> 
> 
> So just Python and Node, both incoming and outgoing, http and any other protocols (layers)?

Do you mean peer or password? We usually only record host/ip and port in `peer`

Reply:
Like for example in sw_requests.py, the following code will store the user and password if I am not mistaken:
```py
    def _sw_request(this: Session, method, url,
                    params=None, data=None, headers=None, cookies=None, files=None,
                    auth=None, timeout=None, allow_redirects=True, proxies=None,
                    hooks=None, stream=None, verify=None, cert=None, json=None):

        from urllib.parse import urlparse
        url_param = urlparse(url)

        ...

        context = get_context()
        with context.new_exit_span(op=url_param.path or "/", peer=url_param.netloc) as span:
```
I thinkthis is repeated across several plugins.

Reply:
We need to review these plugins and if they really did store username/password, we need to remove them explicitly, usually as the name `peer` indicates, we only store host/IP and port into it